components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/activity/GeckoScreenOrientationDelegate.kt

```diff
@@ -0,0 +1,47 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko.activity
+
+import mozilla.components.concept.engine.activity.ActivityOrientationDelegate
+import mozilla.components.support.base.log.logger.Logger
+import org.mozilla.geckoview.AllowOrDeny
+import org.mozilla.geckoview.AllowOrDeny.ALLOW
+import org.mozilla.geckoview.AllowOrDeny.DENY
+import org.mozilla.geckoview.GeckoResult
+import org.mozilla.geckoview.OrientationController
+import java.lang.ref.WeakReference
+
+/**
+ * Default [OrientationController.OrientationDelegate] implementation that delegates both the behavior
+ * and the returned value to a [ActivityOrientationDelegate].
+ */
+internal class GeckoScreenOrientationDelegate(
+    private val delegateRef: WeakReference<ActivityOrientationDelegate>
```

Comment:
Why does the delegate need to be a weak reference? We don't have a direct reference to the activity from here, right? 🤔

Reply:
With the way `ScreenOrientationFeature` works there should be no need for a weak reference. Removed it.